### PR TITLE
python3Packages.pyflipper: init at unstable 2024-04-15

### PR DESCRIPTION
**Pull request by siraben**
## Description of changes

Related https://github.com/NixOS/nixpkgs/issues/81418

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 343729` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pyflipper</li>
    <li>python311Packages.pyflipper.dist</li>
    <li>python312Packages.pyflipper</li>
    <li>python312Packages.pyflipper.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyflipper/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pyserial,
+  websocket-client
```
Comment (fabaff) [inline pkgs/development/python-modules/pyflipper/default.nix:6]:
```suggestion
  websocket-client,
```

File: pkgs/development/python-modules/pyflipper/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "pyflipper";
+  version = "unstable-2024-04-15";
+  format = "setuptools";
```
Comment (fabaff) [inline pkgs/development/python-modules/pyflipper/default.nix:12]:
```suggestion
  pyproject = true;
```

and add `build-system = [ setuptools ];`

File: pkgs/development/python-modules/pyflipper/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub {
+    owner = "wh00hw";
+    repo = "pyFlipper";
+    rev = "e8a82a25eb766fac53a2e6e5fff6505f60cf0897";
+    hash = "sha256-CQ6oVVkLxyoNoe7L0USfal1980VkfiuHc4cqXTsZ2Jc=";
+  };
+
+  propagatedBuildInputs = [ pyserial websocket-client ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pyflipper/default.nix:21]:
```suggestion
  dependencies = [ pyserial websocket-client ];
```

File: pkgs/development/python-modules/pyflipper/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pyserial,
+  websocket-client
+}:
+
+buildPythonPackage rec {
+  pname = "pyflipper";
+  version = "unstable-2024-04-15";
```
Comment (natsukium) [inline pkgs/development/python-modules/pyflipper/default.nix:11]:
```suggestion
  version = "0.18-unstable-2024-04-15";
```
https://github.com/wh00hw/pyFlipper/blob/e8a82a25eb766fac53a2e6e5fff6505f60cf0897/setup.py#L11

**Comment by siraben** [comment]
@fabaff @natsukium added your suggestions.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 343729` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pyflipper</li>
    <li>python311Packages.pyflipper.dist</li>
    <li>python312Packages.pyflipper</li>
    <li>python312Packages.pyflipper.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 343729` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pyflipper</li>
    <li>python311Packages.pyflipper.dist</li>
    <li>python312Packages.pyflipper</li>
    <li>python312Packages.pyflipper.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyflipper/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  pythonImportsCheck = [
+    "pyflipper"
+  ];
+
+  meta = with lib; {
+    description = "Flipper Zero Python CLI Wrapper";
+    homepage = "https://github.com/wh00hw/pyFlipper";
+    license = licenses.mit;
+    maintainers = with maintainers; [ siraben ];
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pyflipper/default.nix:33]:
```suggestion
  meta = {
    description = "Flipper Zero Python CLI Wrapper";
    homepage = "https://github.com/wh00hw/pyFlipper";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ siraben ];
```

**Comment by fabaff** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 343729`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.pyflipper</li>
    <li>python311Packages.pyflipper.dist</li>
    <li>python312Packages.pyflipper</li>
    <li>python312Packages.pyflipper.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Pull request merged**